File: frontend_v2/src/types/journey.ts
// src/types/journey.ts
export interface JourneyResource {
    id: string;
    name: string;
    description?: string;
    type: string;
    mime_type: string;
    size: number;
    created_time: string;
    modified_time: string;
  }
  
  export interface JourneyLevel {
    id: string;
    name: string;
    description: string;
    order: number;
    resources?: JourneyResource[];
  }
  
  export interface Journey {
    levels: JourneyLevel[];
    total_levels: number;
  }

File: backend/app/models/journey.py
# app/models/journey.py

from dataclasses import dataclass
from typing import List, Dict, Any, Optional

@dataclass
class JourneyResource:
    """Structure for journey resource metadata"""
    id: str
    name: str
    description: Optional[str]
    type: str
    mime_type: str
    size: int
    created_time: str
    modified_time: str

@dataclass
class JourneyLevel:
    """Structure for journey level"""
    id: str
    name: str
    description: str
    order: int
    resources: List[JourneyResource]

@dataclass
class Journey:
    """Structure for complete journey"""
    levels: List[JourneyLevel]
    total_levels: int